politeiad/backendv2/tstorebe/plugins/pi/hooks.go

```diff
@@ -390,11 +392,256 @@ func (p *piPlugin) voteSummary(token []byte) (*ticketvote.SummaryReply, error) {
+	// Decode payload
+	var v comments.Vote
+	err := json.Unmarshal([]byte(payload), &v)
+	if err != nil {
+		return err
+	}
+
+	if !isInCommentTree(latestAuthorUpdate.CommentID, v.CommentID, cs) {
+		return backend.PluginError{
+			PluginID:  pi.PluginID,
+			ErrorCode: uint32(pi.ErrorCodeWritesAllowedOnlyOnUpdates),
+		}
+	}
+
+	return nil
+}
+
+// commentNewAllowedOnApprovedProposal verifies that the given new comment
+// is allowed on the proposal which finished voting, it's vote was approved
+// and is associated with the provided token.
+func (p *piPlugin) commentNewAllowedOnApprovedProposal(token []byte, payload string, latestAuthorUpdate comments.Comment, cs []comments.Comment) error {
+	// Decode payload
+	var n comments.New
+	err := json.Unmarshal([]byte(payload), &n)
+	if err != nil {
+		return err
+	}
+
+	// A new comment on an approved proposal must either be an update
+	// from the author (parent ID will be 0) or a reply to the latest
+	// author update.
+	isUpdateReply := isInCommentTree(latestAuthorUpdate.CommentID, n.ParentID, cs)
+	switch {
+	case n.ParentID == 0:
+		// This might be an update from the author.
+		// Get proposal author to ensure new comment's author is
+		// the proposal's author.
+		recordAuthorID, err := p.recordAuthor(token)
+		if err != nil {
+			return err
+		}
+
+		if n.UserID == recordAuthorID &&
+			n.ExtraData != "" && n.ExtraDataHint != "" {
+			switch n.ExtraDataHint {
+			case pi.ProposalUpdateHint:
+				// Decode comment extra data
+				var pum pi.ProposalUpdateMetadata
+				err = json.Unmarshal([]byte(n.ExtraData), &pum)
+				if err != nil {
+					return err
+				}
+				// Verify update title
+				if !p.titleIsValid(pum.Title) {
+					return backend.PluginError{
+						PluginID:     pi.PluginID,
+						ErrorCode:    uint32(pi.ErrorCodeUpdateTitleInvalid),
+						ErrorContext: p.titleRegexp.String(),
+					}
+				}
+				// New valid author update
+				return nil
+
+			default:
+				return backend.PluginError{
+					PluginID:  pi.PluginID,
+					ErrorCode: uint32(pi.ErrorCodeExtraDataHintInvalid),
+				}
+
+			}
+		}
+		// New comment is a new thread but not a valid author update.
+		return backend.PluginError{
+			PluginID:     pi.PluginID,
+			ErrorCode:    uint32(pi.ErrorCodeVoteStatusInvalid),
+			ErrorContext: "vote has ended; comments are locked",
+		}
+
+	case isUpdateReply:
+		// This is a reply to the latest update. This is allowed.
+		return nil
+
+	case !isUpdateReply:
+		// New comment is a reply, but is not a reply to the latest update. This
+		// is not allowed.
+		return backend.PluginError{
+			PluginID:  pi.PluginID,
+			ErrorCode: uint32(pi.ErrorCodeWritesAllowedOnlyOnUpdates),
+		}
+
+	default:
+		// This should not happen
+		return errors.Errorf("unknown comment write state")
+	}
+}
+
+// writesAllowedOnApprovedProposal verifies that the given comment write
```

Comment:
Grammar mistakes throughout this sentence that make it difficult to read.  

Also, why put this on a new line?
` // This includes both comments and comment votes.`

This interupts the flow and makes it harder to read.  You wouldn't see this type of formatting in a blog post or book.